### PR TITLE
Dict SDK Functions

### PR DESCRIPTION
**Pull request by sahilravinder**
Added support for the following Dict Functions

- DictSDK.diff
- DictSDK.intersect
- DictSDK.union
- DictSDK.foldl
- DictSDK.foldr
- DictSDK.map
- DictSDK.merge

### CODE REVIEW

File: morphir/src/org/finos/morphir/runtime/sdk/DictSDK.scala [hunk 1]
```diff
@@ -27,4 +27,121 @@ object DictSDK {
...
       }
   }
+
+  val diff = DynamicNativeFunction2("diff") {
+    // === ELM Function ===
+    // diff : Dict comparable a -> Dict comparable b -> Dict comparable a
+    // Keep a key-value pair when its key does not appear in the second dictionary.
+    (ctx: NativeContext) => (dict1Raw: RTValue.Map, dict2Raw: RTValue.Map) =>
+      {
+        val dict1 = dict1Raw.value.clone()
```
Comment (DamianReeves) [inline morphir/src/org/finos/morphir/runtime/sdk/DictSDK.scala:37]: The use of clone is not needed in this implementation as `filter` will not mutate the original underlying `LinkedHashMap`

Comment (DamianReeves) [reply morphir/src/org/finos/morphir/runtime/sdk/DictSDK.scala:37]: https://www.scala-lang.org/api/2.13.6/scala/collection/mutable/LinkedHashMap.html#filter(pred:A=%3EBoolean):C

File: morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala
[the remark this answers is not in the document]
Comment (sahilravinder) [reply morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala:1]: working on this

File: morphir/src/org/finos/morphir/runtime/sdk/DictSDK.scala (hunk 1 above)
Comment (sahilravinder) [reply morphir/src/org/finos/morphir/runtime/sdk/DictSDK.scala]: fixed

File: morphir/src/org/finos/morphir/runtime/sdk/DictSDK.scala [hunk 2]
```diff
@@ -27,4 +27,121 @@ object DictSDK {
...
+        val dict2 = dict2Raw.value.clone()
+        val diff  = dict1.filter { case (k, _) => !dict2.contains(k) }
+        RTValue.Map(diff)
+      }
+  }
+
+  val intersect = DynamicNativeFunction2("intersect") {
```
[the remark this answers is not in the document]
Comment (sahilravinder) [reply morphir/src/org/finos/morphir/runtime/sdk/DictSDK.scala:42]: fixed

File: morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala
[the remark this answers is not in the document]
Comment (sahilravinder) [reply morphir/runtime/test/jvm/src/org/finos/morphir/runtime/EvaluatorMDMTests.scala:1]: added tests and updated implementation to remove clone

**Review by DamianReeves — APPROVED** [review]
LGTM

**Pull request merged**